### ISSUE TITLE
Issues & disruptions regarding Water's Shield 

### ISSUE THREAD

**Issue by Whaffles** [belathus/eldritch_revolution#18]
TL;DR - The Quest Water's Shield is completed on ALL right clicks and many left clicks; the "Positive Potion Effect" (ID 80) awarded can give VERY BAD potion effects, and makes the game almost unplayable; the "Has Potion Requirement" looks like its broken by a typo; potion effect 78 is Revival, not Sun Scorned (ID 115); the reward for Water's Shield may not be working as intended in general.

Possible solution - fix quest condition and/or redesign quest reward(s)?

---

Full post (sorry for the small essay, but there are a lot of details [and conjecture] regarding this issue):

One of the two possible potion effects given as a reward from the Water's Shield quest is the Not Enough Potions "Random Positive Effect" potion (ID 80). This potion, when applied, selects a random "positive" (any potion effect without the red border) potion effect and applies it to the subject for 60 seconds. The list of possible effects include INCREDIBLY disruptive effects such as "Floating" (ID 47) - which causes the player to float helplessly upwards as if they were attacked by a shulker, ultimately killing anyone unfortunate enough to not have a pond in the area, milk in their inventory, or flight - and "Collapse" (ID 91) - which causes the player's camera, but not their controls, to be inverted (and for some bizarre reason is not coded in as a negative effect). 

These negative effects cause players to be almost entirely unable to play the game for the effect's full duration, and can be applied to the player any time they interact with blocks and tile entities.

Here is some (important) conjecture, though: all that I said above might be irrelevant. The quest Water's Shield does not seem to be working as intended at all. The description suggests that the quest is completed - and the potion effect(s) (more about this detail later) applied - only when the player is not affected by the Sun Scorned potion effect from Thaumcraft. However, the "Has Potion Effect" condition is not checking for the presence of any potions - probably because that potion value was entered into the wrong field. Important side note: the effect the quest seems to be meant to check for (and one of the two potion effects rewarded by the quest) is 78, which is "Revival." Sun Scorned is 115. If this condition is corrected, then the (surprisingly huge) issue regarding this quest would be abated.

**Comment by Whaffles** [comment]
Also, if suggestions are appreciated, turn down the effect duration of the potion reward to 9000 (from 90000) so it is more in line with the other Innsmouth quests, remove potion effect 80 (Random Positive Effect) as a possible reward, and instead reward the player with one effect from a list of your choosing that always applies alongside Sun Scorned. (Currently the quest rewards either the Revival or Random Positive Effect potion effect upon completion - not both). The Random Positive Effect potion effect gives the player a potion effect that lasts 60 seconds, regardless of what duration you assign to it. It sounds like this makes it unsuitable for how you want to use it. Progression allows for a working alternative, luckily (mentioned above).

Lastly, I just want to say the Innsmouth questline is pretty damn awesome (well, in theory - more than one quest in it is broken). Its a cool way of changing up a person's playstyle, by making life on land tricky and rewarding life underwater.

Keep up the good work.

**Comment by Divineaspect** [comment]
Suggestions are appreciated!

Yeah, that was an extraneous set of 00s added to  water's shield.
Should be resolved in 3.20

These are great thoughts and helpful feedback.
Heroes file proper bug reports like these.
